mitodl/ol-data-platform
# Apply deduplication to dbt models for airbyte incremental sync using a timestamp cursor field

### What are the relevant tickets?




https://github.com/mitodl/hq/issues/3665

### Description (What does it do?)

Adding a function to apply deduplication queries to a few open edx models where the source tables should be loaded by airbyte incremental syncs using timestamp cursor field. Details are in https://github.com/mitodl/hq/issues/3665

### How can this be tested?

Since we haven't switched to use timestamp cursor fields for these raw tables yet, the test will just to ensure these models run successfully without error
```
dbt build stg__mitxonline__openedx__mysql__courseware_studentmodule
dbt build stg__mitxpro__openedx__mysql__courseware_studentmodule
dbt build stg__mitxonline__openedx__mysql__grades_subsectiongrade
dbt build stg__mitxonline__openedx__mysql__grades_subsectiongradeoverride
dbt build irx__mitx__openedx__mysql__certificates_generatedcertificate
dbt build irx__mitxonline__openedx__mysql__certificates_generatedcertificate
dbt build irx__xpro__openedx__mysql__certificates_generatedcertificate
dbt build irx__mitx__openedx__mysql__grades_persistentcoursegrade
dbt build irx__mitxonline__openedx__mysql__grades_persistentcoursegrade
dbt build irx__xpro__openedx__mysql__grades_persistentcoursegrade
dbt build irx__mitx__openedx__mysql__grades_persistentsubsectiongrade
dbt build irx__mitxonline__openedx__mysql__grades_persistentsubsectiongrade
dbt build irx__xpro__openedx__mysql__grades_persistentsubsectiongrade
dbt build irx__mitx__openedx__mysql__workflow_assessmentworkflow
dbt build irx__mitxonline__openedx__mysql__workflow_assessmentworkflow
dbt build irx__xpro__openedx__mysql__workflow_assessmentworkflow
```
